proto/osmosis/gamm/v1beta1/genesis.proto

```diff
@@ -21,6 +21,6 @@ option go_package = "github.com/osmosis-labs/osmosis/v10/x/gamm/types";
 message GenesisState {
   repeated google.protobuf.Any pools = 1
       [ (cosmos_proto.accepts_interface) = "PoolI" ];
-  uint64 next_pool_number = 2;
+  uint64 next_pool_id = 2;
```

Comment:
AFAIU, when you break proto like this, you technically have to bump the package version.

Reply:
So field name changes like this break clients and json serialization, but I think its fine to do so for genesis

cc @daniel-farina  @czarcas7ic if field renames / genesis file breaks like this end up being problematic for integrators

Reply:
Actually I'll just add a json serialization name override, so this then won't break anything